masq_lib/src/test_utils/mock_websockets_server.rs

```diff
@@ -192,17 +200,19 @@ impl MockWebSocketsServer {
                                 log(
                                     do_log,
                                     index,
-                                    "Responding to a request for FireAndForget message in dirrection to UI",
+                                    "Responding to a request for FireAndForget message in direction to UI",
                                 );
-                                inner_responses_arc
-                                    .lock()
-                                    .unwrap()
-                                    .clone()
-                                    .into_iter()
-                                    .for_each(|message| {
-                                        client.send_message(&message).unwrap();
-                                        thread::sleep(Duration::from_millis(20))
+                                if let Some(tuple) = self.broadcast_trigger_acessories.take() {
```

Comment:
Suggestion: Modify this logic so that when a `broadcastTrigger` message is received, the Mock responds only with FireAndForget messages. If the queue is empty, or its head is a Conversation message, the Mock would not respond at all; if there are two FireAndForgets, a Conversation, and another FireAndForget, the Mock would send the first two FireAndForgets and stop.

Also: perhaps the `broadcastTrigger` message could be extended to include a numeric parameter to limit the number of FireAndForgets sent in response to that particular trigger. So...if `broadcastTrigger` says, "Send me two!" and there are five waiting in the queue, only two are sent; but if `broadcastTrigger` says "Send me ten!" and there are only five waiting in the queue, only the five are sent: the queue is not further searched for five more.